src/realm/sync/noinst/client_impl_base.hpp

```diff
@@ -158,15 +162,15 @@ class ClientImpl {
     const bool m_fix_up_object_ids;
     const std::function<RoundtripTimeHandler> m_roundtrip_time_handler;
     const std::string m_user_agent_string;
-    network::Service m_service;
-    std::mt19937_64 m_random;
-    websocket::EZSocketFactory m_socket_factory;
+    std::shared_ptr<websocket::DefaultSocketProvider> m_socket_provider;
```

Comment:
This should be the interface instead, so we can stick the custom provider when needed.